pkg/handlers/handlers.go

```diff
@@ -36,23 +47,15 @@ func HandleFile(ctx context.Context, file io.Reader, chunkSkel *sources.Chunk, c
 
 	// Process the file and read all []byte chunks from handlerChan.
 	handlerChan := handler.FromFile(ctx, file)
-	for {
-		select {
-		case data, open := <-handlerChan:
-			if !open {
-				// We finished reading everything from handlerChan.
-				return true
-			}
-			chunk := *chunkSkel
-			chunk.Data = data
-			// Send data on chunksChan.
-			select {
-			case chunksChan <- &chunk:
-			case <-ctx.Done():
-				return false
-			}
-		case <-ctx.Done():
+	for data := range handlerChan {
+		chunk := *chunkSkel
+		chunk.Data = data
+
+		if common.IsDone(ctx) {
 			return false
 		}
+		// Send data on chunksChan.
+		chunksChan <- &chunk
 	}
+	return true
```

Comment:
This refactored version operates slightly differently. Previously the function would return any time the context was canceled, but now it will only return if there is data to read and it's not blocked by the `chunksChan`.

Since either reading from `handlerChan` or writing to `chunksChan` can block, I'm not seeing an obvious way to simplify what we have.